include: storage: flash_map: _NODE macros

michalek-no opened the pull request:
adds size, offset and divice access macros that are operating with nodes as an argumet.

michalek-no committed:
include: storage: flash_map: _NODE macros

adds size, offset and divice access macros that are
operating with nodes as an argumet.

Signed-off-by: Mateusz Michalek <mateusz.michalek@nordicsemi.no>

de-nordic reviewed (commented):
LGTM.
Nevertheless the Flash Map API is considered stable API (https://docs.zephyrproject.org/2.7.5/reference/api/overview.html#api-overview) and therefore tests scenarios are required by the change.

michalek-no committed:
tests: storage: flash_map: _NODE macros

adds API checks for fixed partitions access macros that are
called with nodes as an argumet.

Signed-off-by: Mateusz Michalek <mateusz.michalek@nordicsemi.no>

michalek-no commented:
rebase + tests

kartben commented on include/zephyr/storage/flash_map.h:
```diff
@@ -310,6 +319,15 @@ uint8_t flash_area_erased_val(const struct flash_area *fa);
+ *
+ * @param node DTS node of a partition
+ *
+ * @return fixed-partition offset, as defined for the partition in DTS.
```
```suggestion
 * @return fixed-partition size, as defined for the partition in DTS.
```

kartben commented on tests/subsys/storage/flash_map/src/main.c:
```diff
@@ -190,4 +191,15 @@ ZTEST(flash_map, test_flash_area_erased_val)
 	flash_area_close(fa);
 }
 
+ZTEST(flash_map, test_fixed_partition_noded_macros)
```
```suggestion
ZTEST(flash_map, test_fixed_partition_node_macros)
```

kartben reviewed (changes requested):
couple typos spotted
